hermes/src/store/storage/local_storage.rs

```diff
@@ -155,12 +157,507 @@ impl Storage for LocalStorage {
+            .is_err());
+    }
+
+    pub fn create_empty_accumulator_state_at_slot(slot: Slot) -> AccumulatorState {
+        AccumulatorState {
+            slot,
+            accumulator_messages: None,
+            wormhole_merkle_state: None,
+        }
+    }
+
+    #[tokio::test]
+    pub async fn test_store_and_receive_accumulator_state_works() {
+        // Initialize a storage with a cache size of 2 per key and the accumulator state.
+        let storage = LocalStorage::new_instance(2);
+
+        // Create and store an accumulator state with slot 10.
+        let accumulator_state = create_empty_accumulator_state_at_slot(10);
+
+        // Store the accumulator state.
+        storage
+            .store_accumulator_state(accumulator_state.clone())
+            .await
+            .unwrap();
+
+        // Make sure the retrieved accumulator state is what we stored.
+        assert_eq!(
+            storage.fetch_accumulator_state(10).await.unwrap().unwrap(),
+            accumulator_state
+        );
+    }
+
+    #[tokio::test]
+    pub async fn test_store_and_receive_accumulator_state_works_on_overwrite() {
+        // Initialize a storage with a cache size of 2 per key and the accumulator state.
+        let storage = LocalStorage::new_instance(2);
+
+        // Create and store an accumulator state with slot 10.
+        let mut accumulator_state = create_empty_accumulator_state_at_slot(10);
+
+        // Store the accumulator state.
+        storage
+            .store_accumulator_state(accumulator_state.clone())
+            .await
+            .unwrap();
+
+        // Retrieve the accumulator state and make sure it is what we stored.
+        assert_eq!(
+            storage.fetch_accumulator_state(10).await.unwrap().unwrap(),
+            accumulator_state
+        );
+
+        // Change the state to have accumulator messages
+        // We mutate the existing state because the normal flow is like this.
+        accumulator_state.accumulator_messages = Some(AccumulatorMessages {
+            magic:     [0; 4],
+            slot:      10,
+            ring_size: 3,
+            messages:  vec![],
+        });
+
+        // Store the accumulator state again.
+        storage
+            .store_accumulator_state(accumulator_state.clone())
+            .await
+            .unwrap();
+
+        // Make sure the retrieved accumulator state is what we stored.
+        assert_eq!(
+            storage.fetch_accumulator_state(10).await.unwrap().unwrap(),
+            accumulator_state
+        );
+    }
+
+    #[tokio::test]
+    pub async fn test_store_and_receive_multiple_accumulator_state_works() {
+        // Initialize a storage with a cache size of 2 per key and the accumulator state.
+        let storage = LocalStorage::new_instance(2);
+
+        let accumulator_state_at_slot_10 = create_empty_accumulator_state_at_slot(10);
+        let accumulator_state_at_slot_20 = create_empty_accumulator_state_at_slot(20);
```

Comment:
could also do create_and_store here to reduce code duplication

Reply:
👍